sbc_compassion/models/import_letter_line.py

```diff
@@ -78,10 +79,13 @@ def _set_sponsorship_id(self):
                     [('child_id.code', '=', inst.child_code)]) and
                     inst.env['recurring.contract'].search([
                         ('partner_codega', '=', inst.partner_codega)]))
-
-                inst.sponsorship_id = inst.env['recurring.contract'].search([
+                temp = inst.env['recurring.contract'].search([
                     ('child_id.code', '=', inst.child_code),
                     ('partner_codega', '=', inst.partner_codega)])
+                # check if only one sponsorship is find and,
+                # if not, find the best one
+                temp = self._check_sponsorship(temp)
+                inst.sponsorship_id = temp
```

Comment:
The code below is useless because after `_check_sponsosrship`, the length of sponsorship will always be 1.